Mathlib/Dynamics/TopologicalEntropy/Subset.lean

```diff
@@ -0,0 +1,218 @@
+  · exact h' ▸ le_top
+  obtain ⟨s, s_cover, s_coverMincard⟩ := (coverMincard_finite_iff T F U n).1 h'
+  exact s_coverMincard ▸ (s_cover.closure h V_uni).coverMincard_le_card
+
+lemma coverEntropyInfEntourage_closure (h : Continuous T) (F : Set X) (U : Set (X × X))
+    {V : Set (X × X)} (V_uni : V ∈ 𝓤 X) :
+    coverEntropyInfEntourage T (closure F) (U ○ V) ≤ coverEntropyInfEntourage T F U :=
+  expGrowthInf_monotone fun n ↦ ENat.toENNReal_mono (coverMincard_closure_le h F U V_uni n)
+
+lemma coverEntropyEntourage_closure (h : Continuous T) (F : Set X) (U : Set (X × X))
+    {V : Set (X × X)} (V_uni : V ∈ 𝓤 X) :
+    coverEntropyEntourage T (closure F) (U ○ V) ≤ coverEntropyEntourage T F U :=
+  expGrowthSup_monotone fun n ↦ ENat.toENNReal_mono (coverMincard_closure_le h F U V_uni n)
+
+lemma coverEntropyInf_closure (h : Continuous T) {F : Set X} :
+    coverEntropyInf T (closure F) = coverEntropyInf T F := by
+  refine (iSup₂_le fun U U_uni ↦ ?_).antisymm (coverEntropyInf_monotone T subset_closure)
+  obtain ⟨V, V_uni, V_U⟩ := comp_mem_uniformity_sets U_uni
+  exact le_iSup₂_of_le V V_uni ((coverEntropyInfEntourage_antitone T (closure F) V_U).trans
+    (coverEntropyInfEntourage_closure h F V V_uni))
+
+theorem coverEntropy_closure (h : Continuous T) {F : Set X} :
+    coverEntropy T (closure F) = coverEntropy T F := by
+  refine (iSup₂_le fun U U_uni ↦ ?_).antisymm (coverEntropy_monotone T subset_closure)
+  obtain ⟨V, V_uni, V_U⟩ := comp_mem_uniformity_sets U_uni
+  exact le_iSup₂_of_le V V_uni ((coverEntropyEntourage_antitone T (closure F) V_U).trans
+    (coverEntropyEntourage_closure h F V V_uni))
+
+end Closure
+
+/-! ### Finite unions -/
+
+section Union
+
+lemma IsDynCoverOf.union {T : X → X} {F G : Set X} {U : Set (X × X)} {n : ℕ} {s t : Set X}
+    (hs : IsDynCoverOf T F U n s) (ht : IsDynCoverOf T G U n t) :
+    IsDynCoverOf T (F ∪ G) U n (s ∪ t) :=
+  union_subset (hs.trans (biUnion_subset_biUnion_left subset_union_left))
+    (ht.trans (biUnion_subset_biUnion_left subset_union_right))
+
+lemma coverMincard_union_le (T : X → X) (F G : Set X) (U : Set (X × X)) (n : ℕ) :
+    coverMincard T (F ∪ G) U n ≤ coverMincard T F U n + coverMincard T G U n := by
+  classical
+  rcases eq_top_or_lt_top (coverMincard T F U n) with hF | hF
+  · rw [hF, top_add]; exact le_top
+  rcases eq_top_or_lt_top (coverMincard T G U n) with hG | hG
+  · rw [hG, add_top]; exact le_top
+  obtain ⟨s, s_cover, s_coverMincard⟩ := (coverMincard_finite_iff T F U n).1 hF
+  obtain ⟨t, t_cover, t_coverMincard⟩ := (coverMincard_finite_iff T G U n).1 hG
+  rw [← s_coverMincard, ← t_coverMincard, ← ENat.coe_add]
+  apply (IsDynCoverOf.coverMincard_le_card _).trans (WithTop.coe_mono (s.card_union_le t))
+  rw [s.coe_union t]
+  exact s_cover.union t_cover
+
+lemma coverEntropyEntourage_union {T : X → X} {F G : Set X} {U : Set (X × X)} :
+    coverEntropyEntourage T (F ∪ G) U
+      = coverEntropyEntourage T F U ⊔ coverEntropyEntourage T G U := by
```

Comment:
```suggestion
      = max (coverEntropyEntourage T F U) (coverEntropyEntourage T G U) := by
```
over linear orders, we favor `max` and `min` over the weird lattice notations. Same thing below.